src/test/java/com/synopsys/integration/alert/performance/SlackPerformanceTest.java

```diff
@@ -0,0 +1,96 @@
+
+import org.junit.Ignore;
+import org.junit.jupiter.api.Test;
+
+import com.google.gson.JsonObject;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.common.enumeration.ProcessingType;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.FieldValueModel;
+import com.synopsys.integration.alert.common.rest.model.JobFieldModel;
+import com.synopsys.integration.alert.performance.model.SlackPerformanceProperties;
+import com.synopsys.integration.alert.performance.utility.NotificationWaitJobTask;
+import com.synopsys.integration.wait.WaitJob;
+
+public class SlackPerformanceTest extends IntegrationPerformanceTest {
+    private final static String SLACK_PERFORMANCE_JOB_NAME = "Slack Performance Job";
+    private final SlackPerformanceProperties slackProperties = new SlackPerformanceProperties();
+
+    private String blackDuckProviderID = "-1";
+
+    @Test
+    @Ignore
+    public void testAlertPerformance() throws Exception {
+        LocalDateTime startingTime = LocalDateTime.now();
+        intLogger.info(String.format("Starting time %s", getDateTimeFormatter().format(startingTime)));
+
+        logTimeElapsedWithMessage("Logging in took %s", startingTime, LocalDateTime.now());
+        startingTime = LocalDateTime.now();
+
+        // Create the Black Duck Global provider configuration
+        blackDuckProviderID = setupBlackDuck();
+        
+        logTimeElapsedWithMessage("Configuring the Black Duck provider took %s", startingTime, LocalDateTime.now());
+        startingTime = LocalDateTime.now();
+
+        String jobId = createSlackJob();
+        String message = String.format("Creating the Job %s jobs took", SLACK_PERFORMANCE_JOB_NAME);
+        logTimeElapsedWithMessage(message + " %s", startingTime, LocalDateTime.now());
+
+        LocalDateTime startingSearchDateTime = LocalDateTime.now();
+        // trigger BD notification
+        triggerBlackDuckNotification();
+
+        NotificationWaitJobTask notificationWaitJobTask = new NotificationWaitJobTask(intLogger, getDateTimeFormatter(), getGson(), getAlertRequestUtility(), startingSearchDateTime, jobId);
+        WaitJob waitForNotificationToBeProcessed = WaitJob.create(intLogger, 600, startingSearchDateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli(), 20, notificationWaitJobTask);
+        boolean isComplete = waitForNotificationToBeProcessed.waitFor();
+        intLogger.info("Finished waiting for the notification to be processed: " + isComplete);
+        assertTrue(isComplete);
+    }
+
+    private String createSlackJob() throws Exception {
+        Map<String, FieldValueModel> providerKeyToValues = new HashMap<>();
+        providerKeyToValues.put("provider.common.config.id", new FieldValueModel(List.of(blackDuckProviderID), true));
```

Comment:
Use the constants for the names of the fields that are put into the map from these classes:

- ProviderDescriptor
- ProviderDistributionUIConfig
- ChannelDistributionUIConfig
- SlackDescriptor